tests/test_lane.py

```diff
@@ -449,34 +437,51 @@ def test_apply(self):
         low_quality.quality = 0
         low_quality.random = 1
 
+        self.add_to_materialized_view(
+            [high_quality_low_random, high_quality_high_random,
+             low_quality]
+        )
+
         facets = FeaturedFacets(0.5, False)
-        base_query = self._db.query(Work).join(Work.license_pools)
+        from model import MaterializedWorkWithGenre as work_model
+        base_query = self._db.query(work_model).join(work_model.license_pool)
+
+        def expect(works, qu):
+            expect_ids = [x.works_id for x in qu]
+            actual_ids = [x.id for x in works]
+            eq_(expect_ids, actual_ids)
 
         # Higher-tier works show up before lower-tier works.
         #
         # Within a tier, works with a high random number show up
         # before works with a low random number. The exact quality
         # doesn't matter (high_quality_2 is slightly lower quality
         # than high_quality_1), only the quality tier.
-        featured = facets.apply(self._db, base_query, Work, False)
-        eq_(
+        featured = facets.apply(self._db, base_query, False)
+        expect(
             [high_quality_high_random, high_quality_low_random, low_quality],
-            featured.all()
+            featured
         )
 
         # Switch the random numbers, and the order of high-quality
         # works is switched, but the high-quality works still show up
         # first.
-        high_quality_high_random.random = 0
-        high_quality_low_random.random = 1
-        eq_([high_quality_low_random, high_quality_high_random, low_quality], 
-            featured.all())
+        high_quality_high_random.random = 0.12
+        high_quality_low_random.random = 0.98
+        self._db.commit()
+        SessionManager.refresh_materialized_views(self._db)
+        expect([high_quality_low_random, high_quality_high_random, low_quality], featured)
 
-        # Passing in distinct=True makes the query distinct.
+        # Passing in distinct=True makes the query distinct by work ID.
         eq_(False, base_query._distinct)
-        distinct_query = facets.apply(self._db, base_query, Work, True)
-        eq_(True, distinct_query._distinct)
+        distinct_query = facets.apply(self._db, base_query, True)
+        eq_([work_model.works_id], distinct_query._distinct)
 
+#        # Passing in distinct=True makes the query distinct on
```

Comment:
You commented this test case out.

Reply:
There were two test cases and I went back on forth on which one was the 'right' way to do the code. I've removed the commented-out case in worklist-featured-feed.